lib/src/error/src/ably_exception.dart

```diff
@@ -31,9 +31,14 @@ class AblyException implements Exception {
 
   @override
   String toString() {
+    if (errorInfo != null) {
+      return 'AblyException ($errorInfo)';
```

Comment:
Have you confirmed that if `errorInfo` _is_ not `null` then `code` and `message` _do not_ need to be communicated to the user? Currently the structure/API of this `AblyException` class is a bit confusing - especially given all three fields are extracted from the constructor that takes a `PlatformException`, implying that they all carry discrete informational value. 🤔

Reply:
Good catch, I've thought that PlatformException takes care of that, but it doesn't even hold the type of errorInfo.
52b1c98c20251eeb10d53c51868aae1f257f96c0